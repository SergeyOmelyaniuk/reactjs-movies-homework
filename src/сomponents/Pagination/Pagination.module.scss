@import '../../scss/variables.scss';

.pagination {
	display: flex;
}

.item {
	width: 40px;
	height: 40px;
	font-size: 18px;
	margin-right: 10px;
	border-radius: 50%;
	text-align: center;
	line-height: 40px;
	border: 1px solid transparent;
	cursor: pointer;
	&:hover {
		border: 1px solid $colorGreen;
	}
	&.active {
		background-color: $colorGreen;
	}
}
