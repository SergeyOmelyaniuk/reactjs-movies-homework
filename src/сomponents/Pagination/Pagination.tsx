import React from 'react';
import styles from './Pagination.module.scss';
import classnames from 'classnames';
import getCountPages from '../../helpers/getCountPages';

interface PaginationProps {
	totalPages: number;
	currentPage: number;
	setCurrentPage: (value: number) => void;
}

const Pagination = ({
	totalPages,
	currentPage,
	setCurrentPage,
}: PaginationProps) => {
	const pages = getCountPages(totalPages, currentPage);

	return (
		<ul className={styles.pagination}>
			{pages.map((page) => (
				<li
					key={page}
					className={classnames(styles.item, {
						[styles.active]: currentPage === page,
					})}
					onClick={() => setCurrentPage(page)}
				>
					{page}
				</li>
			))}
		</ul>
	);
};

export default Pagination;
