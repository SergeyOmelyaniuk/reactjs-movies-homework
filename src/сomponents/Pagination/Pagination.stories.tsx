import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Pagination from './Pagination';

export default {
	title: 'Components/Pagination',
	component: Pagination,
	argTypes: {},
} as ComponentMeta<typeof Pagination>;

const Template: ComponentStory<typeof Pagination> = (args) => (
	<div
		style={{
			width: '400px',
			height: '100px',
			backgroundColor: '#202020',
			display: 'flex',
			justifyContent: 'center',
			alignItems: 'center',
			color: 'white',
		}}
	>
		<Pagination {...args} />
	</div>
);

export const Default = Template.bind({});
Default.args = {
	totalPages: 5,
	currentPage: 1,
	setCurrentPage: action('Change page'),
};
