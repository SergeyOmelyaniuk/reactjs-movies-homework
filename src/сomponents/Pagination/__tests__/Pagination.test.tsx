import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Pagination from '..';

describe('Pagination', () => {
	test('should render', () => {
		const { container } = render(<Pagination pages={5} />);

		expect(container).toMatchSnapshot();
	});

	test('should change page', () => {
		const { container, getByText } = render(<Pagination pages={5} />);

		const menuElement = getByText(4);
		fireEvent.click(menuElement);

		expect(container).toMatchSnapshot();
	});
});
