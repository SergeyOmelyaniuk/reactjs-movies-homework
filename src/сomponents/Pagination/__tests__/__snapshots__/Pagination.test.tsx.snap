// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should change page 1`] = `
<div>
  <ul
    class="pagination"
  >
    <li
      class="item"
    >
      1
    </li>
    <li
      class="item"
    >
      2
    </li>
    <li
      class="item"
    >
      3
    </li>
    <li
      class="item active"
    >
      4
    </li>
    <li
      class="item"
    >
      5
    </li>
  </ul>
</div>
`;

exports[`Pagination should render 1`] = `
<div>
  <ul
    class="pagination"
  />
</div>
`;
